client/src/main/python/slipstream/SlipStreamHttpClient.py

```diff
@@ -286,7 +286,8 @@ def get_server_configuration(self):
         return config
 
     def login(self, username, password):
-        self._httpPost(self.serviceurl + '/auth/login', body={
+        self._httpPost(self.authnServiceUrl, body={
+            'href': 'session-template/internal',
```

Comment:
This should be extended to allow any authentication method to be used.  The form needs to have the key "href" and then any keys defined with the referenced template.  For "internal", the keys are "username" and "password"; for "api-key", "key" and "secret".

Reply:
(Not necessarily with this PR.)

Reply:
Yes I'm agree with you.